vignettes/datatable-benchmarking.Rmd

```diff
@@ -17,11 +17,56 @@ h2 {
 }
 </style>
 
-This document is meant to guide on measuring performance of data.table. Single place to documents best practices or traps to avoid.
+This vignette documents the best ways to accurately measure data.table's performance, as well as common traps to avoid.
+
+## `microbenchmark` is not always appropriate
+Repetitive benchmarking like that used by `microbenchmark` may not always be appropriate. 
+While it makes perfect sense for timing atomic calculations, it may provide misleading results for common data processing tasks, which are typically larger than the scale for which `microbenchmark` was designed, and run only once.
+
+For example, the following gives the impression that `data.table` is an order of magnitude slower than `data.frame` for updating columns.
```

Comment:
Again this, together with the example below, creates a bad impression like I'm being gaslighted.
To an untrained eye, the example looks perfectly reasonable. And if you don't explain what exactly is bad with the example, I will probably be thinking in the back of my head: "ok looks like data.table is sometimes orders of magnitudes slower, and in some other example it was only 1.5 times slower (0.02 vs 0.03). So why do they say there's no real difference?"

Btw measuring the second example (n=1e7) with microbenchmark still shows that data.table is more than twice slower on average:
```
> microbenchmark(DF$y <- seq.int(nrow(DF)), DT[, y := .I], times=50L)
Unit: milliseconds
                      expr       min        lq     mean   median       uq
 DF$y <- seq.int(nrow(DF))  8.267302  9.715634 21.92438 19.45919 21.14266
         DT[, `:=`(y, .I)] 17.495159 20.004846 47.44068 45.19089 49.50766
```
And I appreciate the fact that the data is being changed by-reference, so running this code multiple times may overwrite the old data instead of creating a new vector. However I don't see why this would create any perceptible performance difference: allocating a vec is supposed to be extremely fast...
 
Also I just re-run the benchmark for n=1e8, and the speed difference is 3 times, not in data.table's favor. So I don't know, maybe pick a different example? I'm running R 3.4.2 and afair it had some performance improvements to `seq.int`.

Reply:
There are going to be big improvements to `seq` function in next major R release. They won't be materialized unless needed, so better to chose some function that will stay valid for this example.